src/segment-loader.js

```diff
@@ -598,10 +598,10 @@ export default class SegmentLoader extends videojs.EventTarget {
     if (!newPlaylist) {
       return;
     }
-
     const oldPlaylist = this.playlist_;
     const segmentInfo = this.pendingSegment_;
 
+    this.startingMedia_ = void 0;
```

Comment:
On playlist change we have to reset starting media, as the previous media may have been blacklisted and caused this change, so it's startingMedia_ was invalid.